scipy/interpolate/src/__fitpack.cc

```diff
@@ -418,9 +420,152 @@ fpknot(const double *x_ptr, ssize_t m,
     }
 
     // round up, like Dierckx does? This is really arbitrary though.
-    ssize_t idx_newknot = (ix[idx_max] + ix[idx_max+1] + 1) / 2;
+    d_ssize_t idx_newknot = (ix[idx_max] + ix[idx_max+1] + 1) / 2;
 
     return x(idx_newknot);
 }
 
+
+
+/*
+ * Evaluate the spline function
+*/
+void
+_evaluate_spline(
+    const double *tptr, d_ssize_t len_t,         // t, shape (len_t,)
```

Comment:
if these d_ssize_t vars are not changing add `const` to them so compiler can do more magic with them.

Reply:
Gotta love const-correctness :-). Given that these size variables come from `PyArray_DIMS(array, 0)`, which is an `npy_intp` not `const npy_intp`, let me see what sort of hoops I'll need to jump through in the wrapper. Once I figure out the `npy_intp` vs `ptrdiff_t` integer type madness across 32-bit linux, windows and whatever clang is doing on MacOS.

I think I gave up trying to be "idiomatic" with `ptrdiff_t` on the C++ side, `npy_intp` on the Numpy C API side and `dtype=np.intp` in python. The current attempt is to use `int64_t` for all indexing (which is what `d_ssize_t` is typedef-d to), let's see what the CI battery is to say about it. When it deigns to show a green checkmark, I'll see about const-correctness.

Reply:
I'm not comment on C++ things. You dug your own hole you deal with it 😅

Reply:
Nah, that's exactly the same in C, tracing back to K&R : int vs long vs long long types being platform-dependent. 

I _am_ digging my own hole in a different place: I want a clear separation between C (C++, doesn't matter) and the C-to-python glue. This means `__fitpack.{h,cc}` does not  know about `Py_ssize_t` or `npy_intp` types. Then the need to pass an integer array between python and C via numpy C API descends into madness even if I control all of it.

I _suspect_ it will be a bit saner when we only have numpy 2.0, but for now it's... uhm... iterative. 

On a related subject: do you see downsides of using `int64_t` for all indexing? (Size of arrays is not a concern, just buy some more memory, sorry).

Reply:
As long as you don't do BLAS/LAPACK calls I don't think so, it should be good to go.

Reply:
I think it is perfectly fine to use elementary types like `Py_ssize_t` or `npy_intp` in the "non-glue" C/C++ part.